spec/p2p/p2p-top-level.md

```diff
@@ -0,0 +1,329 @@
+      statement](#Informal-Problem-statement): For the general
+      audience, that is, engineers who want to get an overview over what
+      the component is doing from a bird's eye view.
+    - [Sequential Problem statement](#Sequential-Problem-statement):
+      Provides a mathematical definition of the problem statement in
+      its sequential form, that is, ignoring the distributed aspect of
+      the implementation of the blockchain.
+
+- [Part III](#part-iii---as-distributed-system): Distributed
+  aspects, system assumptions and temporal
+  logic specifications.
+
+  - [Incentives](#incentives): how faulty full nodes may benefit from
+    misbehaving and how correct full nodes benefit from cooperating.
+  
+  - [Computational Model](#Computational-Model):
+      timing and correctness assumptions.
+
+  - [Distributed Problem Statement](#Distributed-Problem-Statement):
+      temporal properties that formalize safety and liveness
+      properties in the distributed setting.
+
+- [Part IV](#part-iv---Protocol):
+  Specification of the protocols.
+
+     - [Definitions](#Definitions): Describes inputs, outputs,
+       variables used by the protocol, auxiliary functions
+
+     - [Protocol](#core-verification): gives an outline of the solution,
+       and details of the functions used (with preconditions,
+       postconditions, error conditions).
+
+     - [Liveness Scenarios](#liveness-scenarios): when the light
+       client makes progress depends heavily on the changes in the
+       validator sets of the blockchain. We discuss some typical scenarios.
+
+- [Part V](#part-v---supporting-the-ibc-relayer): Additional
+  discussions and analysis
+
+
+In this document we quite extensively use tags in order to be able to
+reference assumptions, invariants, etc. in future communication. In
+these tags we frequently use the following short forms:
+
+- TMBC: Tendermint blockchain
+- SEQ: for sequential specifications
+- LIVE: liveness
+- SAFE: safety
+- FUNC: function
+- INV: invariant
+- A: assumption
+
+--->
+
+# Part I - A Tendermint node 
+
+TODO: 
+- we should survey here what we think are the expectations of the reactors from p2p? 
+    - consensus might need 
+        - for liveness that neighborhood is stable
+        - proposers are not disconnected?
+    - mempool might just need that a transaction can reach each validator within a reasonable amount of time (this might be achievable if at no point in time the current graph is connected, but, e.g., something along the lines that the union of the graphs over some period in time is connected.)
+    >  talk about the 1-to-1 and the 1-to-many/all delivery guarantees needed by the protocols (do this for each protocol. discuss how these requirements translate into what p2p should guarantee)
+- What does p2p expect from the reactors? (don't falsely report bad nodes; this puts requirements on the reactors and perhaps/likely also on the application running on top of ABCI)
```

Comment:
- to provide reasonable parameters to drive the connections of peers. 
     > These parameters can be low level (like what we have now, i.e., number of outgoing connections, number of incoming connections, permanent peers) or higher level (e.g., form a tree, for a ring with long n fingers...), which require a more elaborate PEX.